telos-frontend/src/views/journal/Page.jsx

```diff
@@ -1,9 +1,75 @@
+import { useState, useCallback, useEffect } from 'react';
+import { useDrop } from 'react-dnd';
+import update from 'immutability-helper';
+import DraggableWidget from '../../dnd/DraggableWidget';
+import { WidgetTypes } from '../../dnd/WidgetTypes';
 import './Page.css';
 
-const Page = () => (
-  <div className="Page">
-    <textarea />
-  </div>
-);
+const Page = ({ date, leftPage }) => {
+  const [widgets, setWidgets] = useState({});
+
+  useEffect(() => {
+    // Temp code to display all widgets. Remove when backend is ready.
+    if (leftPage) {
+      setWidgets({
+        0: { widgetType: 'todo', top: 0, left: 0 },
```

Comment:
What's the reasoning for changing the data format to this style instead of having the id as a field within each widget object?

Reply:
I just thought it would be more efficient rather than iterating through an array to get a single widget. Though probably no difference as there will only be a few widgets each page.